Improve grant entry forms and grant details page

scottjones opened the pull request:
This has turned into a rather large pull request...

Improves the form entry for grants to give not only the criteria and description, but also the amount of the grant requested and details about the rent.  The PDF forms include more information about this right now but I would like for us to learn more about how that actually is used.

This pull request also combines the property and payee steps into one.  It also includes the income sources code (merged from the existing pending work that had already been started).  Additionally, it improves the grant status page.

Crisis and Grant Request are one page (note the new money entry type):
![image](https://cloud.githubusercontent.com/assets/444693/15451348/731631e2-1f83-11e6-8a06-8b1f7c7d0dab.png)

The budget page money entries are different looking:
![image](https://cloud.githubusercontent.com/assets/444693/15436771/27303a8e-1e89-11e6-84c5-695f8b6069ff.png)

Property and Payee are on one page:
![image](https://cloud.githubusercontent.com/assets/444693/15448615/bf96b14c-1f2c-11e6-9c42-172e17aec3fc.png)

Income Sources are now working:
![image](https://cloud.githubusercontent.com/assets/444693/15451355/c85c5366-1f83-11e6-8cf6-763fbef9167f.png)

The grant details (show) page is improved:
![image](https://cloud.githubusercontent.com/assets/444693/15451295/2e2c6bca-1f81-11e6-928e-730d521456c5.png)

Commit:
Temporary changes for adding source of income

Commit:
Merge remote-tracking branch 'origin/issue-216-property-information-page' into temp-source-of-income-page

scottjones committed:
Style the select2 drop downs with the same font

* app/assets/stylesheets/forms.scss: Don't let select2 override our
  font.

scottjones committed:
Combine grant request and criteria pages

* config/locales/en.yml:
* app/controllers/forms_controller.rb: Reduce steps of the wizard by
  one and combine "criteria" and "ask" into overview.

* app/views/forms/ask.html.erb: Remove.

* app/views/forms/criteria.html.erb ->
  app/views/forms/overview.html.erb: Rename.

scottjones committed:
Begin adding additional fields to overview page

* app/controllers/application_controller.rb: Add method for stripping
  values that contain $ or , and are money but we want to use like
  floats.

* app/controllers/forms_controller.rb: Allow money fields to be
  submitted for the grant.

* app/models/subsidy_type.rb: Add a to_s method.

* app/views/forms/overview.html.erb: Add a few new fields and improve
  formatting!

scottjones committed:
Improve money inputs and expand grant entry form

* app/assets/javascripts/money.coffee: Format money without the $ and
  without commas.

* app/controllers/forms_controller.rb: Allow a few more inputs.
  Eliminate whitelisted params.

* app/models/custom_inputs/money_input.rb: New simple form input.

* app/views/budgets/_budget_fields.html.erb: Use new simple form input
  for money.

* app/views/forms/overview.html.erb: Use new simple form input for
  money and add the fields for total amount owed and the grant amount
  requested.

* app/views/grants/show.html.erb: Change icon for the budget
  worksheet.

* config/initializers/simple_form.rb: Register the CustomInputs
  namespace for custom simple forms inputs.

* config/initializers/simple_form_bootstrap.rb: Add wrapper for inline
  fields for the budget.

* db/migrate/20160520001116_grant_total_amount_owed_and_requested_amount.rb:
* db/schema.rb: Add requested amount and total amount owed.

scottjones committed:
Merge remote-tracking branch 'origin/master' into improve-crisis-and-grant-request-forms

scottjones committed:
Simplify budget commit

* app/controllers/application_controller.rb: Remove strip_money method
  as it's no longer used.

* app/controllers/budgets_controller.rb: Remove lots of code around
  handling the money input fields for budget purposes.

scottjones committed:
Combine property and payee pages

* app/assets/stylesheets/layout.scss: Make h2 and hr closer together
  if back to back.

* app/controllers/forms_controller.rb: Remove step for payee and add
  some more allowed params.

* app/views/forms/applicants.html.erb:
* app/views/forms/overview.html.erb: Shift column offset to 3/9.
  Adjust headers.

* app/views/forms/payee.html.erb: Remove.

* app/views/forms/properties.html.erb: Add payee information.  Arrange
  form horizontally.  Put address and unit, city and state on the same
  row.

* config/initializers/simple_form_bootstrap.rb: Add a few more wrapper
  options to set up forms how we want.

* config/locales/en.yml: Remove payees step and rename properties
  step.

scottjones committed:
Correct indentation on simple_form wrapper.

scottjones committed:
Add grant coverage field for intended use.

* app/controllers/forms_controller.rb: One more allowable field for
  grant usage.

* app/models/coverage_type.rb: Add a to_s method.

* app/views/forms/overview.html.erb: Add form field for Grant Usage,
  and change label for Grant Amount Requested (to simply be "Grant
  Requested").

scottjones committed:
Property/Payee zip code and payee phone should be guarded as required

* app/views/forms/properties.html.erb: Guard zip codes and payee phone
  number fields as required.

scottjones committed:
Merge remote-tracking branch 'origin/master' into temp-source-of-income-page

scottjones committed:
Rubocop fix for hash alignment

scottjones committed:
Arrange income fields better.  Make them save too!

* app/assets/stylesheets/forms.scss: Add a horizontal line between
  employment and the person/employment fields above.

* app/controllers/forms_controller.rb: Correct typo on the
  incomes_attributes.

* app/views/forms/fields/_income_fields.html.erb: Fix layout and
  date fields.

* app/views/forms/fields/_person_fields.html.erb: Include already
  saved incomes.  Move income fields outside of the person fields.

scottjones committed:
Merge remote-tracking branch 'origin/temp-source-of-income-page' into improve-crisis-and-grant-request-forms

scottjones committed:
Keep income on its own page.  Too long of a form.

* app/assets/stylesheets/forms.scss: Remove employment style as we
  don't need it anymore.

* app/assets/stylesheets/layout.scss: Add some extra styles for
  headings and hrs.

* app/models/person.rb: Add a to_s method.

* app/views/forms/employment.html.erb: Loop over each person and offer
  adding of employment for each one.

* app/views/forms/fields/_income_fields.html.erb: Move start and end
  date up side by side on medium and higher screens.

* app/views/forms/fields/_person_fields.html.erb: Remove income
  fields.

* config/locales/en.yml: Change "Sources of Income" to "Income
  Sources"

scottjones committed:
Don't show duplicate applications when they have mulitple people

* app/models/grant.rb: Don't join people, include them.  Tweak the
  search function to ensure we get all the applicants from a grant in
  our search results.

* spec/models/grant_spec.rb: Add shared examples for searching.  Make
  sure when we search that we find the FULL grant with all applicants,
  not just the matching applicant.

scottjones committed:
Remove redundant "self" and format search query more nicely.

* app/models/grant.rb: Remove redundant "self" and format the search
  on multiple lines.

scottjones committed:
Adjust how "Add Employment" button looks.

* app/assets/stylesheets/layout.scss: Reduce space below our
  underlines (combo of h2, etc. + hr).  Add a little padding to the
  side of glyphicons.

* app/views/forms/employment.html.erb: Place add employment button
  next to the applicant rather than below the employment.

scottjones committed:
Remove instructions from grant request form

* app/views/forms/overview.html.erb: These instructions are incomplete
  and overly wordy.

scottjones committed:
Improve the "grant show" page some.

* app/assets/stylesheets/layout.scss: h3 should use a normal weight.

* app/views/grants/show.html.erb: Remove lots of labels.  Arrange a
  little more nicely.  Add a few buttons.

scottjones committed:
Remove gray color from glyphicons.

* app/assets/stylesheets/layout.scss: This was affecting even buttons,
  etc.  Remove color from glyphicons.

lamplus commented on app/controllers/forms_controller.rb:
```diff
@@ -56,7 +56,7 @@ def payees_attributes
   def residence_attributes
     { residence_attributes: [:id, :residence_type_id, :address, :unit_number, :city,
-                             :state, :zip, :_destroy] }
+                             :state, :zip, :move_in_date, :_destroy] }
```
Good catch.

scottjones committed:
Merge branch 'master' into improve-crisis-and-grant-request-forms

lamplus commented on app/views/forms/fields/_person_fields.html.erb:
```diff
@@ -1,3 +1,5 @@
+ <% income_fields = 'forms/fields/income_fields' %>
```
Is it needed?

scottjones replied:
Good catch; No, it's not...

lamplus replied:
I will finish the rest tonight.
On May 21, 2016 6:41 PM, "Scott Jones" notifications@github.com wrote:

> In app/views/forms/fields/_person_fields.html.erb
> https://github.com/on-site/grantzilla/pull/265#discussion_r64142776:
> 
> > @@ -1,3 +1,5 @@
> > - <% income_fields = 'forms/fields/income_fields' %>
> 
> Good catch; No, it's not...
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> https://github.com/on-site/grantzilla/pull/265/files/de90b99910bdc8613d88bc3b694eee0556647b43#r64142776

scottjones committed:
Make HIF grant controls for money look the same

* app/assets/javascripts/components/GrantControlsForm.jsx: Add $ input
  add-on.

* app/views/grants/_controls.html.erb: Include grant payee correctly.

scottjones committed:
Fixes #255: Adds required fields

* app/views/forms/fields/_person_fields.html.erb: Require email
  address for applicants.

* app/views/forms/overview.html.erb: Require grant usage type.

* app/views/forms/properties.html.erb: Require residence type.

scottjones committed:
Remove extra variable declaration referencing income per PR.

* app/views/forms/fields/_person_fields.html.erb: Old variable
  declaration is no longer needed now that we're not including income
  when entering applicants.

lamplus commented on app/views/grants/show.html.erb:
```diff
@@ -1,90 +1,76 @@
+            <%= @grant.user.full_name %>
+          </h4>
+          <div>
+            <span class="glyphicon glyphicon-earphone"></span>
```
Should the phone icon be displayed only if phone is available?

lamplus commented on app/views/grants/show.html.erb:
```diff
@@ -1,90 +1,76 @@
+            <% end %>
+          </div>
+          <div>
+            <span class="glyphicon glyphicon-envelope"></span>
```
Should the email icon be displayed only if email is available?

lamplus commented on app/views/grants/show.html.erb:
```diff
@@ -1,90 +1,76 @@
-              </p>
+              <h4 class="light"><%= person.full_name %></h4>
+              <div>
+                <%= time_ago_in_words(person.birth_date) %> old
```
Need to check person.birth_date.present? before using time_ago_in_words.

lamplus commented:
Reviewed up to 104cee6

scottjones committed:
Don't show the phone icon if no phone number is on file per PR

* app/views/grants/show.html.erb: Change the checks for presence of
  the headers and phone, and ensure the header and phone icon are not
  shown unless the value is present.

scottjones committed:
Don't attempt to show age if there's no birth date

* app/views/grants/show.html.erb: Check for the birth date presense
  before computing the age of the applicant.
